ctx/ctx.go

```diff
@@ -133,3 +136,29 @@ func GetDefinition(r *http.Request) *apidef.APIDefinition {
 	}
 	return nil
 }
+
+func SetOASDefinition(r *http.Request, s *oas.OAS) {
+	ctx := r.Context()
+	ctx = context.WithValue(ctx, OASDefinition, s)
+	setContext(r, ctx)
+}
+
+func GetOASDefinition(r *http.Request) *oas.OAS {
+	if v := r.Context().Value(OASDefinition); v != nil {
+		if val, ok := v.(*oas.OAS); ok {
```

Comment:
Can we put the insides here into something like `func cloneOAS(*oas.OAS) (*oas.OAS, error)`;

- I'd like a benchmark of that function so we're in a place to replace it / optimize it (maybe encoding/gob is faster to make a copy),
- I'd like to unit test a modification of an internal value (map, slice, pointer) and assert it's a copy